drivers/hud_spm_report/spec/models/datalab_testkit/all_projects_spec.rb

```diff
@@ -50,112 +51,93 @@
+    end
+    it 'Measure 4.3' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '4.3',
+      )
+    end
+    it 'Measure 4.4' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '4.4',
+      )
+    end
+    it 'Measure 4.5' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '4.5',
+      )
+    end
+    it 'Measure 4.6' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '4.6',
+        # off by one
+        skip: ['C3', 'C4'],
+      )
+    end
 
-  #   it 'Measure 5.1' do
-  #     compare_results(
-  #       file_path: result_file_prefix + 'spm',
-  #       question: '5.1',
-  #       # Pending AAQ: [SPM] All Projects - Data lab test kit Measures 2 and 5 - historic data missing - submitted 10/4/2022
-  #       skip: [
-  #         'C3',
-  #         'C4',
-  #       ],
-  #     )
-  #   end
+    it 'Measure 5.1' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '5.1',
+      )
+    end
 
-  #   it 'Measure 5.2' do
-  #     compare_results(
-  #       file_path: result_file_prefix + 'spm',
-  #       question: '5.2',
-  #       # Pending AAQ: [SPM] All Projects - Data lab test kit Measures 2 and 5 - historic data missing - submitted 10/4/2022
-  #       skip: [
-  #         'C3',
-  #         'C4',
-  #       ],
-  #     )
-  #   end
+    it 'Measure 5.2' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '5.2',
+        # off by 10
+        skip: ['C2', 'C4'],
+      )
+    end
 
-  #   it 'Measure 7a.1' do
-  #     compare_results(
-  #       file_path: result_file_prefix + 'spm',
-  #       question: '7a.1',
-  #       # Not included in test file since it's an internal calculation
-  #       skip: [
-  #         'C5',
-  #       ],
-  #     )
-  #   end
+    it 'Measure 7a.1' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '7a.1',
+      )
+    end
 
-  #   it 'Measure 7b.1' do
-  #     compare_results(
-  #       file_path: result_file_prefix + 'spm',
-  #       question: '7b.1',
-  #       # Not included in test file since it's an internal calculation
-  #       skip: [
-  #         'C4',
-  #       ],
-  #     )
-  #   end
+    it 'Measure 7b.1' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '7b.1',
+        # off by 3
+        skip: ['C2', 'C3', 'C4'],
+      )
+    end
 
-  #   it 'Measure 7b.2' do
-  #     compare_results(
-  #       file_path: result_file_prefix + 'spm',
-  #       question: '7b.2',
-  #       # Not included in test file since it's an internal calculation
-  #       skip: [
-  #         'C4',
-  #       ],
-  #     )
-  #   end
+    it 'Measure 7b.2' do
+      compare_results(
+        file_path: result_file_prefix + 'spm',
+        question: '7b.2',
+        # off by 30
```

Comment:
Not sure about the source of these extra records. Something I notice is that 23 of the extra records have entry_date > move_in_date; all of the expected records have entry_date <= move_in_date. I don't see direction in this in the spec except as it relates to "General Report Instructions: household member joins the household after they are already housed" which should already be accounted for in `SpmEnrollment.move_in_date`

Reply:
Entry dates must be <= move-in.  If they are after move-in, the move-in is considered invalid and should not be used.  Maybe that's what's going on?  If a household member joins a household after the move-in of the HoH, their move-in date is their entry date.

Reply:
Good to know, that seems to have resolved the issue and 7b.2 now matches testkit. Thanks!

Reply:
Excellent!